### PR TITLE
Add headers and status code to mockable content in NSURLSession mocks

### LINKED ISSUE #2: Mock headers in a response instead of just the body

### PR DESCRIPTION
**Pull request by kerrmarin**
Fixes #2 and #3

### CODE REVIEW

File: Example/Tests/NSURLSessionTests.swift [hunk 1]
```diff
@@ -13,12 +13,17 @@ import NSURLSession_Mock
 private class SessionTestDelegate: NSObject, NSURLSessionDataDelegate {
     var expectations: [XCTestExpectation]
     
-    var dataKeyedByTask = Dictionary<Int, NSMutableData>() // [ taskIdentifier: data from task ]
+    var dataKeyedByTask : [Int : NSMutableData] = [:]
+    var response : NSURLResponse?
```
Comment (nap-sam-dean) [inline Example/Tests/NSURLSessionTests.swift]: Shouldn't you need `responseKeyedByTask` here?

File: Example/Tests/NSURLSessionTests.swift [hunk 2]
```diff
@@ -158,4 +163,41 @@ class NSURLSessionTests: XCTestCase {
         }
     }
 
+    
+    func testSession_WithStatusCodeAndHeaders_ShouldReturnTheCorrectStatusCodes() {
+        let expectation = self.expectationWithDescription("Complete called for headers and status code")
+        
+        // Tell NSURLSession to mock thhis URL, each time with different data
```
Comment (FrancescoTatulloNAP) [inline Example/Tests/NSURLSessionTests.swift]: "thhis"

File: Example/Tests/NSURLSessionTests.swift [hunk 3]
```diff
@@ -158,4 +163,41 @@ class NSURLSessionTests: XCTestCase {
...
+        let URL = NSURL(string: "https://www.example.com/1")!
+        let body = "Test response 1".dataUsingEncoding(NSUTF8StringEncoding)!
+        let request = NSURLRequest.init(URL: URL)
+        let headers = ["Content-Type" : "application/test", "Custom-Header" : "Is custom"]
+        NSURLSession.mockSingle(request, body: body, headers: headers, statusCode: 200)
+
+        // Create a session
+        let conf = NSURLSessionConfiguration.defaultSessionConfiguration()
+        let delegate = SessionTestDelegate(expectations: [ expectation ])
```
Comment (FrancescoTatulloNAP) [inline Example/Tests/NSURLSessionTests.swift]: why the spaces? around "expectation"

File: Example/Tests/NSURLSessionTests.swift [hunk 4]
```diff
@@ -158,4 +163,41 @@ class NSURLSessionTests: XCTestCase {
...
+        task.resume()
+        
+        // Validate that the mock data was returned
+        self.waitForExpectationsWithTimeout(1) { timeoutError in
+            XCTAssertNil(timeoutError)
+            
+            XCTAssertEqual(delegate.dataKeyedByTask[task.taskIdentifier], body)
+            guard let response = delegate.response as? NSHTTPURLResponse else {
+                XCTFail("Response isn't the correct type")
+                return
```
Comment (FrancescoTatulloNAP) [inline Example/Tests/NSURLSessionTests.swift:193]: will execution ever get to "return" after an XCTFail?

File: Example/Tests/NSURLSessionTests.swift [hunk 5]
```diff
@@ -158,4 +163,41 @@ class NSURLSessionTests: XCTestCase {
...
+            }
+            XCTAssertEqual(response.statusCode, 200)
+            guard let responseHeaders = response.allHeaderFields as? [String : String] else {
+                XCTFail("Response headers couldn't be transformed to String")
+                return
```
Comment (FrancescoTatulloNAP) [inline Example/Tests/NSURLSessionTests.swift:198]: same as above

File: Example/Tests/NSURLSessionTests.swift (hunk 3 above)
Comment (kerrmarin) [inline Example/Tests/NSURLSessionTests.swift]: Copy-paste, I'll remove them :)

**Pull request merged**